polyfill/test/zoneddatetime.mjs

```diff
@@ -0,0 +1,603 @@
+      it('zdt.millisecond is 123', () => equal(zdt.millisecond, 123));
+      it('zdt.microsecond is 456', () => equal(zdt.microsecond, 456));
+      it('zdt.nanosecond is 789', () => equal(zdt.nanosecond, 789));
+      it('zdt.epochSeconds is 217178610', () => equal(zdt.epochSeconds, 217178610));
+      it('zdt.epochMilliseconds is 217178610123', () => equal(zdt.epochMilliseconds, 217178610123));
+      it('zdt.epochMicroseconds is 217178610123456n', () => equal(zdt.epochMicroseconds, 217178610123456n));
+      it('zdt.epochNanoseconds is 217178610123456789n', () => equal(zdt.epochNanoseconds, 217178610123456789n));
+      it('zdt.dayOfWeek is 4', () => equal(zdt.dayOfWeek, 4));
+      it('zdt.dayOfYear is 323', () => equal(zdt.dayOfYear, 323));
+      it('zdt.weekOfYear is 47', () => equal(zdt.weekOfYear, 47));
+      it('zdt.daysInWeek is 7', () => equal(zdt.daysInWeek, 7));
+      it('zdt.daysInMonth is 30', () => equal(zdt.daysInMonth, 30));
+      it('zdt.daysInYear is 366', () => equal(zdt.daysInYear, 366));
+      it('zdt.monthsInYear is 12', () => equal(zdt.monthsInYear, 12));
+      it('zdt.inLeapYear is true', () => equal(zdt.inLeapYear, true));
+      it('zdt.offset is +00:00', () => equal(zdt.offset, '+00:00'));
+      it('string output is 1976-11-18T15:23:30.123456789+00:00[UTC]', () =>
+        equal(`${zdt}`, '1976-11-18T15:23:30.123456789+00:00[UTC]'));
+    });
+
+    it('casts time zone', () => {
+      const zdt = new ZonedDateTime(epochNanos, 'Asia/Seoul');
+      equal(typeof zdt.timeZone, 'object');
+      assert(zdt.timeZone instanceof Temporal.TimeZone);
+      equal(zdt.timeZone.id, 'Asia/Seoul');
+    });
+    it('defaults to ISO calendar', () => {
+      const zdt = new ZonedDateTime(epochNanos, tz);
+      equal(typeof zdt.calendar, 'object');
+      assert(zdt.calendar instanceof Temporal.Calendar);
+      equal(zdt.calendar.id, 'iso8601');
+    });
+    it('casts calendar', () => {
+      const zdt = new ZonedDateTime(epochNanos, Temporal.TimeZone.from('Asia/Tokyo'), 'japanese');
+      equal(typeof zdt.calendar, 'object');
+      assert(zdt.calendar instanceof Temporal.Calendar);
+      equal(zdt.calendar.id, 'japanese');
+    });
+  });
+
+  describe('string parsing', () => {
+    it('parses with an IANA zone', () => {
+      const zdt = ZonedDateTime.from('2020-03-08T01:00-08:00[America/Los_Angeles]');
+      equal(zdt.toString(), '2020-03-08T01:00:00-08:00[America/Los_Angeles]');
+    });
+    it('parses with an IANA zone but no offset', () => {
+      const zdt = ZonedDateTime.from('2020-03-08T01:00[America/Los_Angeles]');
+      equal(zdt.toString(), '2020-03-08T01:00:00-08:00[America/Los_Angeles]');
+    });
+    it('parses with an IANA zone but no offset (with disambiguation)', () => {
+      const zdt = ZonedDateTime.from('2020-03-08T02:30[America/Los_Angeles]', { disambiguation: 'earlier' });
+      equal(zdt.toString(), '2020-03-08T01:30:00-08:00[America/Los_Angeles]');
+    });
+  });
```

Comment:
No worries. The addition and subtraction tests will be particularly helpful. I didn't include that test because it doesn't pass yet, it will show up in the next PR.